apps/docs/functions/src/rag/hybrid-search.ts

```
@@ -0,0 +1,351 @@
+    topK: topK * 2, // Get more results to allow for filtering
+    minScore: 0.3,
+    category,
+  });
+
+  // Re-score based on keyword presence
+  const rescoredResults = results.map((result) => {
+    const contentLower = result.content.toLowerCase();
+    const titleLower = result.title.toLowerCase();
+
+    let keywordScore = 0;
+    let matchCount = 0;
```

ESLint

[error] 'matchCount' is assigned a value but never used. Allowed unused vars must match /^_/u.

(no-unused-vars)